src/Microsoft.Extensions.Configuration/ConfigurationProvider.cs

```diff
@@ -53,10 +58,40 @@ public virtual void Set(string key, string value)
         /// <summary>
         /// Loads (or reloads) the data for this provider.
         /// </summary>
-        public virtual void Load()
+        public void Load()
         {
+            try
+            {
+                DoLoad();
+            }
+            catch (Exception e)
+            {
+                bool ignoreException = false;
+                if (OnLoadException != null)
+                {
+                    var exceptionContext = new LoadExceptionContext
+                    {
+                        Provider = this,
+                        Exception = e
+                    };
+                    OnLoadException.Invoke(exceptionContext);
+                    ignoreException = exceptionContext.Ignore;
+                }
+                if (!ignoreException)
+                {
+                    throw e;
+                }
+            }
         }
-       
+
+        /// <summary>
+        /// Loads (or reloads) the data for this provider.
+        /// </summary>
+        public virtual void DoLoad()
```

Comment:
So if we are going to promote the delegate to IConfigurationSource, we can actually do this cleaner now via a `LoadSafe` extension method to IConfigurationProvider, we'll need to add the Source property to IConfigurationProvider, but with that we'll be able to move this logic into an extension method instead of doing naming tricks/wrapping a virtual Load.  Breaking changes on the interfaces instead of mucking with Load :)